bundles/com.espressif.idf.serial.monitor/src/com/espressif/idf/serial/monitor/handlers/SerialMonitorHandler.java

```diff
@@ -38,7 +36,7 @@ public Process invokeIDFMonitor()
 		{
 			return monitor.start();
 		}
-		catch (IOException e)
+		catch (Exception e)
 		{
 			Logger.log(e);
 		}
```

Comment:
What do you think about adding the popup message in this catch, that will tell the user, what is going wrong exactly? We can use text like this: "The WebSocket dependency is missing and cannot be installed automatically". Otherwise, it can be confusing for the user, why the IDF monitor is not working, and maybe not be obvious to check the log

Reply:
One suggestion - as this is a core/non-UI class, we could refrain from adding UI dialog here. Instead, we can propagate back the error to the handler who is calling and show the error message in the final UI invoker.

Reply:
Can we create a separate ticket for this I can do it in this but I am afraid it may increase the testing scenarios and will also need to change the code in 2 classes SerialConnector.java and SerialPortHandler.java as one is connecting the monitor and the other one is resuming it in case it is interrupted. So it gets to have multiple scenarios for tests like making sure that both sides are handling exceptions from the open method in the later class invoking the monitor.